include/ddsrouter/event/SignalHandler.hpp

```diff
@@ -129,7 +136,22 @@ class SignalHandler : public EventHandler<int>
     static std::vector<SignalHandler*> active_handlers_;
 
     //! Guards access to variable \c active_handlers_
-    static std::mutex active_handlers_mutex_;
+    static std::recursive_mutex active_handlers_mutex_;
+
+    //! Handle of thread dedicated to listening for signal arrival
+    static std::thread signal_handler_thread_;
+
+    //! Flag used to terminate \c signal_handler_thread_
+    static std::atomic<bool> signal_handler_active_;
+
+    //! Flag set to true when a signal arrives
+    static std::atomic<bool> signal_received_;
```

Comment:
Use a counter to get the number of signals arrived
```suggestion
    static std::atomic<uint32_t> signal_received_;
```